proxy/src/telemetry/metrics/histogram.rs

```diff
@@ -0,0 +1,126 @@
+use std::{cmp, fmt, iter, slice};
+use std::num::Wrapping;
+use std::marker::PhantomData;
+
+use super::Counter;
+
+/// A series of latency values and counts.
+#[derive(Debug, Clone)]
+pub struct Histogram<V: Into<u64>> {
+    buckets: Vec<(Bucket, Counter)>,
```

Comment:
When I made a similar change to `Histogram`s in #775, I implemented this by having the `Histogram` own an array of counts and a reference to an array of bounds (rather than storing the bounds in every instance of `Histogram`):
https://github.com/runconduit/conduit/blob/30dd1a3b8bee68fe18af20d79195b422f84c9df9/proxy/src/telemetry/metrics/histogram.rs#L7-L16

 Including the bounds in each instance effectively doubles the size of the histogram. Since I'd expect we'll create a great deal of `Histogram` instances that all share the same set of bounds, I think having a reference to the bounds will likely lead to a significant difference in memory usage.

This introduces a lifetime parameter, but if the array of bounds is a static array, that's not really a problem.

Reply:
Also, TIOLI, since we don't expect either the bounds or counts to grow over the lifetime of the histogram, con sider `Box<[...]>` rather than `Vec<...>`.

I don't believe there's any real performance difference between these types (someone correct me if I'm wrong), but a boxed array better reflects that we need the storage to be heap-allocated (because it's dynamically sized) but that we don't expect it to grow. When I see a `Vec`, I take that to mean "this will contain a variable number of elements".